synapse/config/cas.py

```diff
@@ -30,7 +30,13 @@ def read_config(self, config, **kwargs):
 
         if self.cas_enabled:
             self.cas_server_url = cas_config["server_url"]
-            self.cas_service_url = cas_config["service_url"]
+            public_base_url = cas_config.get("service_url") or self.public_baseurl
+            if public_base_url[-1] != "/":
+                public_base_url += "/"
```

Comment:
This will change `self.public_baseurl`, which may end up confusing people?

Reply:
It should not since strings are immutable: `public_base_url` doesn't end up as a reference to `self.public_baseurl` unless there's additional magic here that I don't know?

Reply:
Also in reality this code won't run if `self.public_baseurl` is used:

https://github.com/matrix-org/synapse/blob/0cd2938bc854d947ae8102ded688a626c9fac5b5/synapse/config/server.py#L167-L168